node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeadmin/NodeAdminStateUpdater.java

```diff
@@ -128,9 +128,15 @@ private void updateHardwareDivergence(StorageMaintainer maintainer) {
         if (currentState != RESUMED) return;
 
         try {
-            String hardwareDivergence = maintainer.getHardwareDivergence();
-            NodeAttributes nodeAttributes = new NodeAttributes().withHardwareDivergence(hardwareDivergence);
-            nodeRepository.updateNodeAttributes(dockerHostHostName, nodeAttributes);
+            ContainerNodeSpec nodeSpec = nodeRepository.getContainerNodeSpec(dockerHostHostName)
+                    .orElseThrow(() -> new RuntimeException("Failed to get host's node spec from node-repo"));
+            String hardwareDivergence = maintainer.getHardwareDivergence(nodeSpec);
+
+            // Only update hardware divergence if there is a change.
+            if (!nodeSpec.hardwareDivergence.orElse("null").equals(hardwareDivergence)) {
```

Comment:
The benefit of avoiding an update is lost since an additional node repo lookup is added. How about moving this close to fetchContainersToRunFromNodeRepository to utilize that method's lookup?

Reply:
It doesn't really matter since this is only run once and hour, the reason I have is mainly so that we avoid having `PATCH` entries in access-log when nothing has changed.